src/jit/decomposelongs.cpp

```diff
@@ -964,20 +975,252 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
             unreached();
     }
 
-    GenTreeArgList* argList = m_compiler->gtNewArgList(loOp1, hiOp1, shiftWidthOp);
+    // If we are shifting by a constant int, we do not want to save to a helper, instead, we decompose.
+    if (shiftOper == GT_CNS_INT)
```

Comment:
Can't all the helper choosing code above be pushed into the "else" case that actually creates the helper call?